vkrunner/vr-pipeline.c

```diff
@@ -681,15 +688,27 @@ create_vk_layout(struct vr_pipeline *pipeline,
         return layout;
 }
 
-static VkDescriptorSetLayout
+static void
 create_vk_descriptor_set_layout(struct vr_pipeline *pipeline,
                                 const struct vr_script *script)
 {
         struct vr_vk *vkfn = &pipeline->window->vkfn;
         VkResult res;
+        struct vr_buffer desc_sets;
+        struct desc_set_bindings_info *info;
+        size_t n_desc_sets;
         size_t n_buffers = script->n_buffers;
         VkDescriptorSetLayoutBinding *bindings =
                 vr_calloc(sizeof (*bindings) * n_buffers);
+        unsigned prev_desc_set = script->buffers[0].desc_set;
+
```

Comment:
No. In the [line 731](https://github.com/Igalia/vkrunner/pull/12/files/24847f07f18723065c5f74c49ae8758c74810136#diff-e45b40913d7f4c6e7216890831e4187aR731), if the first `prev_desc_set` is `UINT_MAX`, `prev_desc_set != buffer->desc_set` is `true` even when the current `buffer->desc_set` is not the new one.

Reply:
My point was that if you remove the code to add the first entry in `desc_set` so that `desc_set` starts out empty, then when `i == 0` in the for-loop it will add the first entry using the existing code within the for-loop because `buffer->desc_set` will never be `UINT_MAX`. Ie, if `prev_desc_set` is `UINT_MAX` then it is supposed to be a new one.

Reply:
`prev_desc_set` exists to trigger the execution of following lines

```C
                if (prev_desc_set != buffer->desc_set) {
                        info[n_desc_sets].n_bindings =
                                &bindings[i] - info[n_desc_sets].bindings;
                        ++n_desc_sets;
                        info[n_desc_sets].desc_set = buffer->desc_set;
                        info[n_desc_sets].bindings = &bindings[i];
                        prev_desc_set = buffer->desc_set;
                }
```

When `i == 0`, it should not be triggered.
If `prev_desc_set` is initialized as `UINT_MAX`, even when `i == 0`, it triggers the above lines.

Reply:
Set the initial `prev_desc_set` as `UINT_MAX`.
I am not sure the new code is what you intended.
Please review it.